jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -136,6 +138,208 @@ ecma_builtin_typedarray_prototype_length_getter (ecma_value_t this_arg) /**< thi
+                                         ecma_value_t arg2) /**< thisArg */
+{
+  return routines_for_every_some_foreach (this_arg, arg1, arg2, TYPEDARRAY_ROUTINE_EVERY);
+} /* ecma_builtin_typedarray_prototype_every */
+
+/**
+ * The %TypedArray%.prototype object's 'some' routine
+ *
+ * See also:
+ *          ES2015, 22.2.3.9
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_some (ecma_value_t this_arg, /**< this argument */
+                                        ecma_value_t arg1, /**< callbackfn */
+                                        ecma_value_t arg2) /**< thisArg */
+{
+  return routines_for_every_some_foreach (this_arg, arg1, arg2, TYPEDARRAY_ROUTINE_SOME);
+} /* ecma_builtin_typedarray_prototype_some */
+
+/**
+ * The %TypedArray%.prototype object's 'forEach' routine
+ *
+ * See also:
+ *          ECMA-262 v5, 15.4.4.18
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_for_each (ecma_value_t this_arg, /**< this argument */
+                                            ecma_value_t arg1, /**< callbackfn */
+                                            ecma_value_t arg2) /**< thisArg */
+{
+  return routines_for_every_some_foreach (this_arg, arg1, arg2, TYPEDARRAY_ROUTINE_FOREACH);
+} /* ecma_builtin_typedarray_prototype_for_each */
+
+/**
+ * The %TypedArray%.prototype object's 'map' routine
+ *
+ * See also:
+ *          ES2015, 22.2.3.8
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_map (ecma_value_t this_arg, /**< this argument */
+                                       ecma_value_t arg1, /**< callbackfn */
+                                       ecma_value_t arg2) /**< thisArg */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  if (!ecma_op_is_callable (arg1))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Callback function is not callable."));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
+  uint32_t len = ecma_typedarray_get_length (obj_p);
+  ecma_object_t *func_object_p = ecma_get_object_from_value (arg1);
+  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+  ecma_value_t new_typedarray = ecma_op_create_typedarray_with_type_and_length (obj_p, len);
+  ecma_object_t *new_obj_p = ecma_get_object_from_value (new_typedarray);
+
+  for (uint32_t index = 0; index < len && ecma_is_value_empty (ret_value); index++)
+  {
+    ecma_value_t current_index =  ecma_make_uint32_value (index);
+    ecma_value_t get_value = ecma_op_typedarray_get_index_prop (obj_p, index);
+    ecma_value_t call_args[] = { get_value, current_index, this_arg };
```

Comment:
Who frees the get_value? I only see uint8 array tests below, perhaps a float32 array would be good as well.

Same for current_index. You could use ecma_fast_free() here.